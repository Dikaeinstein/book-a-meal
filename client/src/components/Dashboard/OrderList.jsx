import React from 'react';
import PropTypes from 'prop-types';
import Order from './Order';

const OrderList = ({ orders }) => {
  const renderTableBody = () => {
    const orderRows = orders.map(order => (
      <Order key={order.id} order={order} />
    ));
    return (
      <tbody>
        {orderRows}
      </tbody>
    );
  };

  return (
    <div className="orders">
      <table>
        <thead>
          <tr>
            <th>Order ID</th>
            <th>Order Name</th>
            <th>Customer</th>
            <th>Order Amount (&#x20a6;)</th>
            <th>Quantity (plate)</th>
            <th>Total (&#x20a6;)</th>
            <th>Status</th>
            <th>Order Actions</th>
          </tr>
        </thead>
        {renderTableBody()}
      </table>
    </div>
  );
};

OrderList.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default OrderList;
