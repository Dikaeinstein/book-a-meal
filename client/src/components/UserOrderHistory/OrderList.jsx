import React from 'react';
import PropTypes from 'prop-types';
import ConnectedUserOrder from './UserOrder';

const OrderList = ({ orders }) => {
  const renderTableBody = () => {
    const orderRows = orders.map(order =>
      <ConnectedUserOrder key={order.id} order={order} />);
    return (
      <tbody>
        {orderRows}
      </tbody>
    );
  };

  return (
    <div className="orders">
      <table>
        <thead>
          <tr>
            <th>Order ID</th>
            <th>Order Name</th>
            <th>Order Amount (&#x20a6;)</th>
            <th>Quantity</th>
            <th>Total (&#x20a6;)</th>
            <th>Status</th>
            <th>Actions</th>
          </tr>
        </thead>
        {renderTableBody()}
      </table>
    </div>
  );
};

OrderList.propTypes = {
  orders: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default OrderList;
