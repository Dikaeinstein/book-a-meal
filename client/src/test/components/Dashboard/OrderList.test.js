import React from 'react';
import { shallow } from 'enzyme';
import OrderList from '../../../components/Dashboard/OrderList';

describe('<OrderList />', () => {
  const orders = [
    {
      id: 11,
      quantity: 1,
      total: '4000',
      status: 'delivered',
      expired: false,
      mealId: 3,
      userId: 4,
      meal: {
        id: 3,
        name: 'Amala and Ewedu',
        price: '4000',
      },
      user: {
        id: 4,
        name: 'Chinonso Okorie',
      },
    },
  ];
  it('match should snapshot', () => {
    const tree = shallow(<OrderList orders={orders} />);
    expect(tree).toMatchSnapshot();
  });
});
