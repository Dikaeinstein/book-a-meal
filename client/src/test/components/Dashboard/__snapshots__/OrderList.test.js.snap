// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderList /> match should snapshot 1`] = `
<div
  className="orders"
>
  <table>
    <thead>
      <tr>
        <th>
          Order ID
        </th>
        <th>
          Order Name
        </th>
        <th>
          Customer
        </th>
        <th>
          Order Amount (₦)
        </th>
        <th>
          Quantity (plate)
        </th>
        <th>
          Total (₦)
        </th>
        <th>
          Status
        </th>
        <th>
          Order Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <Connect(Component)
        key="11"
        order={
          Object {
            "expired": false,
            "id": 11,
            "meal": Object {
              "id": 3,
              "name": "Amala and Ewedu",
              "price": "4000",
            },
            "mealId": 3,
            "quantity": 1,
            "status": "delivered",
            "total": "4000",
            "user": Object {
              "id": 4,
              "name": "Chinonso Okorie",
            },
            "userId": 4,
          }
        }
      />
    </tbody>
  </table>
</div>
`;
